src/dto/update-appointment.dto.ts

```diff
@@ -1,15 +1,4 @@
-import { IsDateString, IsNotEmpty, IsString } from 'class-validator';
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateAppointmentDto } from './create-appointment.dto';
 
-export class UpdateAppointmentDto {
-  @IsNotEmpty()
-  @IsDateString()
-  start_date: string;
-
-  @IsNotEmpty()
-  @IsDateString()
-  end_date: string;
-
-  @IsNotEmpty()
-  @IsString()
-  phone_number: string;
-}
+export class UpdateAppointmentDto extends PartialType(CreateAppointmentDto) {}
```

Comment:
If we want to allow for partial updates (I'm okay with it 😄) we should update our controller for updating the entity from PUT to PATCH, to hint the API consumers that this endpoint allows for partial updates